controller/Session.class.php

```diff
@@ -31,6 +33,9 @@ public static function init()
             static::setNamespace(static::NAMESPACE_DEFAULT, $oldSession);
         }
 
+        $response = LBRY::logWebVisitor('lbry.io', $_SESSION[static::USER_ID], $_SERVER['REMOTE_ADDR']);
+        $_SESSION[static::USER_ID] = $response['data']['visitor_id'];
```

Comment:
As written, if this code ever breaks in a weird way (e.g. a typo in the above data check), would it end up creating a lot of users server side due to relying on empty session user id to create new users?

Reply:
So I tested this out, by replacing `visitor_id` with `viitor_id` and it actually causes the site to error for all users. So I added `key_exists` checks ( thanks @lyoshenka ) for these. Also you are right that it will endlessly create visitors on internal-apis. 

So I just made a change to link site visitors by ip address as well. If an ip address calls the api twice in 24 hours they will be regarded as the same site visitor. So at most, if this happens it will not create site visitors in internal-apis.

Are you ok with this approach?

Reply:
Sounds sensible to me. Are you/we sure want this called on every request? If so, it might be sensible to do it after the response is flushed to the user (as-is this will presumably add 100+ms to each page render), but this doesn't need to block shipping.

Reply:
I just checked the code and we currently do not do any output buffering.

So if you wanted to pursue above it'd look something like this.

- Add new method `Response::addPostRenderCallback(callable $func)`. This method stores these callables in internal Response array.
- Add new method `Response::invokePostRenderCallbacks()`, which invokes the callbacks.
- Update `Session::init()` to call `Response::addPostRenderCallback` wrapping the existing logic in an inline function.
- Update `index.php` to flush (I think one of and/or both `ob_flush()` or just `flush()`) and then call the above method.

Reply:
I did what you said. Everything on Response is static. So I left it static. Is everything in PHP in the context of a single user?

Reply:
Yes.